services/shhext/service.go

```diff
@@ -235,23 +235,44 @@ func (s *Service) retrieveMessagesLoop(tick time.Duration, cancel <-chan struct{
 				log.Error("failed to retrieve raw messages", "err", err)
 				continue
 			}
+			var messageIDs []string
+
+			for _, messages := range chatWithMessages {
+				for _, message := range messages {
+					messageIDs = append(messageIDs, message.ID.String())
+				}
+			}
+
+			existingMessages, err := s.messenger.MessagesExist(messageIDs)
+			if err != nil {
+				log.Error("failed to chek existing messages", "err", err)
+				continue
+			}
 
 			var signalMessages []*signal.Messages
 
 			for chat, messages := range chatWithMessages {
-				var retrievedMessages []*whisper.Message
-				for _, message := range messages {
-					whisperMessage := message.TransportMessage
-					whisperMessage.Payload = message.DecryptedPayload
-					retrievedMessages = append(retrievedMessages, whisperMessage)
-				}
 
 				signalMessage := &signal.Messages{
 					Chat:     chat,
 					Error:    nil, // TODO: what is it needed for?
-					Messages: s.deduplicator.Deduplicate(retrievedMessages),
+					Messages: s.deduplicator.Deduplicate(messages),
+				}
+
+				var newMessages []dedup.DeduplicateMessage
+
+				// Remove already saved messages
```

Comment:
The comment says we're removing messages, but the method below is appending messages, which is a bit confusing. Maybe the comment could be clearer?